feat(logs): emit usage metrics to app_metrics2 for billing and UI vis…

jonmcwest opened the pull request:
## Problem

Logs product needs usage tracking for billing and rate limiting visibility.

## Changes

### Plugin server (`logs-ingestion-consumer.ts`)

Emits 6 metrics per team per batch to `app_metrics2`:
- `bytes_received` / `records_received` - total incoming
- `bytes_ingested` / `records_ingested` - allowed through rate limiter
- `bytes_dropped` / `records_dropped` - rate limited

### Billing (`usage_report.py`)

Added `logs_bytes_in_period` and `logs_records_in_period` to usage reports, querying `bytes_ingested` and `records_ingested` from `app_metrics2`.

## How did you test this code?

- Added unit + integration tests all passing
- Verified metrics in `app_metrics2` locally: `received = ingested + dropped`

jonmcwest committed:
feat(logs): emit usage metrics to app_metrics2 for billing and UI visibility

Adds app metrics emission to logs ingestion consumer, mirroring how CDP
tracks usage. Emits 6 metrics per team per batch:
- bytes_received / records_received (total incoming)
- bytes_ingested / records_ingested (allowed through rate limiter)
- bytes_dropped / records_dropped (rate limited)

The ingested metrics are already wired into usage_report.py for billing.
The received/dropped metrics enable rate limiting visibility in the UI,
similar to CDP's monitoring dashboard.

Metrics are emitted as a background task alongside log production,
with O(n) aggregation in the existing filter loops to avoid overhead.

benjackwhite commented on plugin-server/src/logs-ingestion/logs-ingestion-consumer.ts:
```diff
@@ -162,6 +198,50 @@ export class LogsIngestionConsumer {
+        if (count === 0) {
+            return Promise.resolve()
+        }
+        return this.kafkaProducer!.produce({
```
Unless I'm missing something this will use the configured producer which currently points at Warpstream.

This needs to point at the main MSK one, so we need to configure two producers and use the correct ones for the different targets.

github-actions[bot] committed:
test(backend): update query snapshots

jonmcwest committed:
feat(logs): add MSK producer config for app_metrics2

Adds KAFKA_MSK_PRODUCER_* env vars to route app_metrics to MSK
while keeping logs data on Warpstream.

jonmcwest committed:
fix(logs): handle metric emission failures gracefully

Updated the logs ingestion consumer to use `Promise.allSettled` for emitting usage metrics. This change ensures that failures in emitting metrics do not block the ingestion process. Added logging to warn about any failed metrics, providing visibility into the number of failures compared to total metrics attempted.

jonmcwest committed:
fix(logs): improve error logging for metric emission failures

jonmcwest committed:
fix(kafka): update KafkaConfigTarget to include METRICS_PRODUCER instead of MSK_PRODUCER
